### PR TITLE
Add a repro for the issue 16663

### LINKED ISSUE #16663: Dashboard filters can stay in URL query string after changing dashboards via search dropdown
**Describe the bug**
If you are on a dashboard with filters, then use the global search drop down suggestion to go to another dashboard that doesn't have any filters, the filters of the first dashboard are retained in the URL query string.

**Logs**
Nothing in either log that appears relevant.

**To Reproduce**
1. Go to any dashboard with filters that have default values, note the URL has filters in the query string.
2. Use the global search to locate another dashboard that doesn't have any parameters, **do not press enter** to navigate to the search results page.
3. From the list of suggestions in the dropdown, click that no filters dashboard.
4. Note the dashboard number in the URL has changed, but the URL now incorrectly includes the previous dashboards filters in the query string.

**Expected behavior**
Dashboards with no parameters shouldn't have any in the URL query string.

**Screenshots**
Filters as expected on the first dashboard and in the URL
image

Use global search to find the other dashboard with no filters (1), click it in the dropdown to navigate (2).
image

Filters incorrectly retained in the URL of the second dashboard which doesn't have any filters
image

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:89.0) Gecko/20100101 Firefox/89.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "1.8.0_161-b12",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_161",
    "java.vm.name": "Java HotSpot(TM) Client VM",
    "java.vm.version": "25.161-b12",
    "os.name": "Windows Server 2012 R2",
    "os.version": "6.3",
    "user.language": "en",
    "user.timezone": "Pacific/Auckland"
  },
  "metabase-info": {
    "databases": [
      "sqlserver",
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "10.5"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.39.2",
      "date": "2021-05-17",
      "branch": "release-x.39.x",
      "hash": "b0d9436"
    },
    "settings": {
      "report-timezone": "Pacific/Auckland"
    }
  }
}
```
Same behaviour noted in Chrome.

**Severity**
Doesn't impact functionality at all from what I can see. If you navigate in the same manner to another dashboard with filters the URL is correct and filters are applied as expected.

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/16663

### Description

The issue doesn't exist in the master branch, most likely it was fixed after introduction of tabs.
This PR contains only a test, which I couldn't run against 0.46.5, but at the same time following the steps manually would broke the test expectations. 

### How to verify

e2e tests should pass. If you want to, download 0.46.5 and follow the steps from the issue

### Demo

https://www.loom.com/share/cee788f2d2bd4a74b6b222afad5071ce

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      ({ body: dashboardCard }) => {
+        const { dashboard_id } = dashboardCard;
+
+        visitDashboard(dashboard_id);
+      },
+    );
+
+    editDashboard();
+
+    cy.get("main header").find(".Icon-gear").click();
```
Comment (WiNloSt) [inline e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js:48]: could you try if 
```suggestion
    dashboardHeader().icon("gear").click();
```
would work? I think using selectors makes the test hard to follow.

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      .findByText("No default")
+      .click();
+
+    popover().contains("Q1").click();
+
+    saveDashboard();
+
+    cy.url().should("include", "quarter_and_year=Q1");
+
+    visitDashboard(1);
```
Comment (kamilmielnik) [inline e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js:60]: :x: 

> This PR contains only a test, which I couldn't run against 0.46.5, but at the same time following the steps manually would broke the test expectations.

It's because `visitDashboard` will reload the page (`cy.visit()`) and the bug is about SPA navigation.

In this test we have to navigate to another dashboard the same way a user would - e.g. clicking a bookmarked dashboard in the sidebar.

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    cy.get("main header").find(".Icon-gear").click();
+
+    cy.findByLabelText("subscriptions sidebar")
+      .findByText("No default")
+      .click();
+
+    popover().contains("Q1").click();
+
+    saveDashboard();
```
Comment (WiNloSt) [inline e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js:56]: All these actions to set the default parameter value could be avoid by setting the default value in dashboard parameter when creating the dashboard directly.

```diff
const parameters = [
  {
    name: "Quarter and Year",
    slug: "quarter_and_year",
    id: "f8ae0c97",
    type: "date/quarter-year",
    sectionId: "date",
+   default: "Q1-2023",
  },
];
```

This should speed up the test. From my machine, it reduces the time from 8-9s to 6s. That's not huge by any mean, but we should avoid using the UI to prepare the data when possible.

**Review by kamilmielnik — CHANGES_REQUESTED** [review]
The test does not reproduce the issue.

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js (hunk 1 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js]: I updated code to use `icon`, but dashboardHeader is 3 levels down comparing to what I need, so it didn't work

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js (hunk 3 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js]: changed, thanks!

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js (hunk 2 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js]: indeed, great catch! I was confused by not seeing search filter in cypress probably because some not loaded css. should be fixed now

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        visitDashboard(dashboard_id);
+      },
+    );
+
+    editDashboard();
+
+    cy.get("main header").icon("gear").click();
+
+    saveDashboard();
```
Comment (WiNloSt) [inline e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js:52]:
```suggestion
```

These 3 actions are probably unnecessary now given we map the parameter default value since the dashboard creation.

Could you try removing them and see if the test still works?

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    cy.url().should("include", "quarter_and_year=Q1");
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js]: it's not necessary, but I did it

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js (hunk 4 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js]: indeed, with default parameter those are not needed, thanks

File: e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+const dashboardDetails = { parameters };
+
+describe("issue 16663", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should remove filter value from url after going to another dashboard (metabase#16663)", () => {
+    const dahsboardToRedirect = "Orders in a dashboard";
```
Comment (kamilmielnik) [inline e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js:33]:
```suggestion
    const dashboardToRedirect = "Orders in a dashboard";
```

**Review by kamilmielnik — APPROVED** [review]
:+1:

**Review by WiNloSt — APPROVED** [review]
awesome!
